File: ruby_small_problems/medium2/triangle_sides.rb
def triangle(s1, s2, s3)
  sides = [s1, s2, s3]

  case
  when 2 * sides.max >= sides.sum, sides.any?(0)
    :invalid
  when sides[0] == sides[1] && sides[1] == sides[2]
    :equilateral
  when sides[1..2].any? { |side| sides[0] == side }
    :isosceles
  else
    :scalene
  end
end

p triangle(3, 3, 3) == :equilateral
p triangle(3, 3, 1.5) == :isosceles
p triangle(3, 4, 5) == :scalene
p triangle(0, 3, 3) == :invalid
p triangle(3, 1, 1) == :invalid

File: javascript_small_problems/medium_problems2/triangle_sides.js
function triangle(side1, side2, side3) {
  let sides = [side1, side2, side3].sort((a, b) => a - b);;

  if (sides[0] === 0) return 'invalid';
  if (sides[0] + sides[1] < sides[2]) return 'invalid';
  if (sides[0] === sides[1] && sides[1] === sides[2]) return 'equilateral';
  if (sides[0] === sides[1] || sides[1] === sides[2] || sides[2] === sides[0]) return 'isosceles'
  return 'scalene';
}

console.log(triangle(3, 3, 3));        // "equilateral"
console.log(triangle(3, 3, 1.5));      // "isosceles"
console.log(triangle(3, 4, 5));        // "scalene"
console.log(triangle(0, 3, 3));        // "invalid"
console.log(triangle(3, 1, 1));        // "invalid"